projects/laji/src/app/+project-form/results/biomon-result/biomon-result.component.ts

```diff
@@ -52,12 +64,47 @@ export class BiomonResultComponent implements OnInit, OnDestroy {
         this.router.navigate([], { queryParams: { tab: Tabs.statistics } });
       }
     });
+    this.taxonOptions$ = this.getTaxonOptions$();
   }
 
   ngOnDestroy(): void {
     this.defaultTabSubscription.unsubscribe();
   }
 
+  getTaxonOptions$(): Observable<{ label: string; value: string }[]> {
```

Comment:
This would be better if it wouldn't have the `subscribe` in the middle since it is not guaranteed that it will always be run immediately and can cause bugs later. It could be refactored like this for example: 
```
    return this.state$.pipe(switchMap(state => (
      this.formApi.getAllForms().pipe(
        rxjsMap(forms =>
          forms
            .filter(f => (f.collectionID !== undefined && f.options.prepopulateWithTaxonSets !== undefined))
            .map(f => ({ collectionID: f.collectionID, taxonSet: f.options.prepopulateWithTaxonSets }))
        ),
        switchMap(pairs => {
          const taxonSet = pairs.find(p => p.collectionID === (state.collection ? state.collection : this.form.collectionID)).taxonSet;
          return this.getOptionsByTaxonSet$(taxonSet);
        })
      )
    )));
```